commit 18b8a9c5dab35c3f39402e12664da6d8b2d52c43
Author: Roopa Prabhu <roopa@cumulusnetworks.com>
Date:   Wed Sep 12 13:21:48 2018 -0700

    net: rtnl_configure_link: fix dev flags changes arg to __dev_notify_flags
    
    [ Upstream commit 56a49d7048703f5ffdb84d3a0ee034108fba6850 ]
    
    This fix addresses https://bugzilla.kernel.org/show_bug.cgi?id=201071
    
    Commit 5025f7f7d506 wrongly relied on __dev_change_flags to notify users of
    dev flag changes in the case when dev->rtnl_link_state = RTNL_LINK_INITIALIZED.
    Fix it by indicating flag changes explicitly to __dev_notify_flags.
    
    Fixes: 5025f7f7d506 ("rtnetlink: add rtnl_link_state check in rtnl_configure_link")
    Reported-By: Liam mcbirnie <liam.mcbirnie@boeing.com>
    Signed-off-by: Roopa Prabhu <roopa@cumulusnetworks.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/core/rtnetlink.c b/net/core/rtnetlink.c
index e3f743c141b3..bafaa033826f 100644
--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -2760,7 +2760,7 @@ int rtnl_configure_link(struct net_device *dev, const struct ifinfomsg *ifm)
 	}
 
 	if (dev->rtnl_link_state == RTNL_LINK_INITIALIZED) {
-		__dev_notify_flags(dev, old_flags, 0U);
+		__dev_notify_flags(dev, old_flags, (old_flags ^ dev->flags));
 	} else {
 		dev->rtnl_link_state = RTNL_LINK_INITIALIZED;
 		__dev_notify_flags(dev, old_flags, ~0U);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1021_18b8a9c5dab35c3f39402e12664da6d8b2d52c43.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1021_18b8a9c5dab35c3f39402e12664da6d8b2d52c43.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
